nasa/earthdata-search
# EDSC-4174: Add metric event to determine when a user adds a collection or a granule to a project

# Overview

### What is the feature?
As part of our metrics objective we want to track when/how users are adding collections and granules to their projects. So that using google analytics we can make informed decisions about usage patterns to drive future feature development of the application.

### What is the Solution?

Created two new events on the middleware one for when granules are added to the project with the UI and one for when collections are added through the UI. These event handler are passed as functions through components to the controllers and expose new GTM data layers that can be picked up by tag manager/GA

We also noticed some of the values for the `defaultClick` event were undefined because the elements target value for the names was undefined added some more places where we can derive the title of the element

### What areas of the application does this impact?

Middleware, Adding new functions on existing `onClicks` which should not affect user interaction in anyway

# Testing

### Reproduction steps

- **Environment for testing:*Any*
- **Collection to test with:*Any*

1. Using Google Tag manager use the preview mode to observe that events are being fired when the following actions are triggered on the UI
2. Click the Add collection to project button that should fire an event repeat that on the table view which should fire the same event with a different payload
3. Click on a collection any, and hit the Add collection button on the granules page that should fire the same collection event but, with a different payload
4. On the granules page for a collection add granule that should fire a **different** event repeat that on the table view that should fire that same `addGranuleToProject` event but, with a different payload

### Attachments
Calls (omitting the gtm-id)
![image](https://github.com/user-attachments/assets/88be373b-4d41-417d-9107-91e552bcd7d0)
![image](https://github.com/user-attachments/assets/3a6b5ab2-4400-41f2-a0a2-9e929a568c71)

# Checklist
